Index: guvcview/data/guvcview.desktop.in.in
===================================================================
--- guvcview.orig/data/guvcview.desktop.in.in
+++ guvcview/data/guvcview.desktop.in.in
@@ -3,9 +3,13 @@ _Name=guvcview
 _GenericName=GTK UVC video viewer
 _X-GNOME-FullName=GTK UVC video viewer
 _Comment=A video viewer and capturer for the linux uvc driver
+Name[pt_BR]=Webcam
+Comment[pt_BR]=Tire fotos e vídeos com sua Webcam.
 TryExec=guvcview
+Keywords=photo;video;webcam;camera;selfie;
+Keywords[pt_BR]=fotos;photos;vídeos;webcam;câmera;selfie;autorretrato;
 Exec=guvcview
-Icon=@prefix@/share/pixmaps/guvcview/guvcview.png
+Icon=org.gnome.Cheese
 Terminal=false
 Type=Application
 Categories=Video;AudioVideo;
