Index: guvcview/guvcview/video_capture.c
===================================================================
--- guvcview.orig/guvcview/video_capture.c
+++ guvcview/guvcview/video_capture.c
@@ -1286,7 +1286,7 @@ void *capture_loop(void *data)
 			render_frame_osd(frame->yuv_frame);
 
 			/* finally render the frame */
-			snprintf(render_caption, 29, "Guvcview  (%2.2f fps)",
+			snprintf(render_caption, 29, "Guvcview",
 				v4l2core_get_realfps(my_vd));
 			render_set_caption(render_caption);
 			render_frame(frame->yuv_frame);
