Index: guvcview/gview_v4l2core/v4l2_core.c
===================================================================
--- guvcview.orig/gview_v4l2core/v4l2_core.c
+++ guvcview/gview_v4l2core/v4l2_core.c
@@ -1241,10 +1241,12 @@ v4l2_frame_buff_t *v4l2core_get_frame(v4
 
 				ret = xioctl(vd->fd, VIDIOC_DQBUF, &vd->buf);
 
-				if(!ret)
+				if(!ret){
 					qind = process_input_buffer(vd);
-				else
+				}else{
 					fprintf(stderr, "V4L2_CORE: (VIDIOC_DQBUF) Unable to dequeue buffer: %s\n", strerror(errno));
+					exit(-1);
+				}
 			}
 			else res = -1;
 
